sdk/iot/hub/tests/cmocka/test_az_iot_hub_client.c

```diff
@@ -24,17 +25,19 @@
 #define TEST_HUB_HOSTNAME_STR "myiothub.azure-devices.net"
 #define TEST_MODULE_ID "my_module_id"
 #define TEST_USER_AGENT "os=azrtos"
+#define PLATFORM_USER_AGENT "DeviceClientType=c/" AZ_SDK_VERSION_STRING
 
 static const az_span test_device_id = AZ_SPAN_LITERAL_FROM_STR(TEST_DEVICE_ID_STR);
 static const az_span test_hub_hostname = AZ_SPAN_LITERAL_FROM_STR(TEST_HUB_HOSTNAME_STR);
 
 static const char test_correct_user_name[]
-    = "myiothub.azure-devices.net/my_device/?api-version=2018-06-30";
+    = "myiothub.azure-devices.net/my_device/?api-version=2018-06-30&" PLATFORM_USER_AGENT;
```

Comment:
Nit: I used to compose the "correct" test strings with a bunch of macros which I also used as the test params. After a little coaxing from @CIPop, I changed my mind to have test params and "correct" output to be unconnected. It provides an opportunity to sort of "double check" to make sure that its right instead of using the exact same value in param and output. So here, it would have whatever the `PLATFORM_USER_AGENT` spelled out in the string literal.

Reply:
yea, but this changes per build because of PLATFORM_USER_AGENT contains the sdk version